From f6c22c8944c02e08eb7d1849ddfe868cd4f8defd Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Wed, 2 Apr 2014 20:51:04 +0200
Subject: [PATCH 117/118] BUG/MEDIUM: pattern: fix wrong definition of the
 pat_prune_fcts array

Commit 6f7203d ("MEDIUM: pattern: add prune function") introduced an
array of functions pat_prune_fcts[] but unfortunately declared it in
pattern.h without marking it "extern", resulting in each file including
it having its own copy.
---
 include/types/pattern.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/types/pattern.h b/include/types/pattern.h
index 3859eda..fa29274 100644
--- a/include/types/pattern.h
+++ b/include/types/pattern.h
@@ -220,7 +220,7 @@ extern char *pat_match_names[PAT_MATCH_NUM];
 extern int (*pat_parse_fcts[PAT_MATCH_NUM])(const char *, struct pattern *, char **);
 extern int (*pat_index_fcts[PAT_MATCH_NUM])(struct pattern_expr *, struct pattern *, char **);
 extern void (*pat_delete_fcts[PAT_MATCH_NUM])(struct pattern_expr *, struct pat_ref_elt *);
-void (*pat_prune_fcts[PAT_MATCH_NUM])(struct pattern_expr *);
+extern void (*pat_prune_fcts[PAT_MATCH_NUM])(struct pattern_expr *);
 extern struct pattern *(*pat_match_fcts[PAT_MATCH_NUM])(struct sample *, struct pattern_expr *, int);
 extern int pat_match_types[PAT_MATCH_NUM];
 
-- 
1.8.3.2

